src/DynamoCore/Linting/Rules/LinterRule.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Linting.Interfaces;
+
+namespace Dynamo.Linting.Rules
+{
+    /// <summary>
+    /// Base class for all linting rules
+    /// </summary>
+    public abstract class LinterRule
+    {
+        /// <summary>
+        /// Unique id of this rule
+        /// </summary>
+        public abstract string Id { get; }
```

Comment:
Let's use the Guid class here and everywhere we need to deal with an id.

Reply:
Can do it for the this LinterRule if we prefer that, on the `LinterExtensionBase` which i think is the only other place we use an Id the string type is dictated by `IExtension`

Reply:
we can leave it as it is, it's similar to many other places